### ISSUE TITLE
Write tconnect data to Sugarmate

### ISSUE THREAD

**Issue by ariana-wolfkiel** [jwoglom/tconnectsync#33]
Sugarmate has no ability to automatically get pump data.

Writing pump data into Sugarmate obtained from tconnect would allow remote monitoring of the X2 in Sugarmate without care providers manually putting in data and provide currently unavailable data from automatic boluses.

Example of usefulness:
We have 4 different care providers who can monitor blood sugar and pump actions via Sugarmate. Different care providers (school med-tech, daycare, relatives) manually input pump information into Sugarmate when they are with the child allowing other providers (parents) to stay informed and respond as necessary with guidance.

X2 runs ControlIQ and does automatic boluses. If remote monitoring (by parents) does not know what boluses are being done automatically, over-bolusing (stacking?) can occur.

Would this be a totally new tool 'tconnectsync-to-Sugarmate' or an additional capability of tconnectsync?

**Comment by jwoglom** [comment]
Hi @ariana-wolfkiel --

I haven't used Sugarmate before, but this should definitely be possible. Right now tconnectsync is only designed to output data to Nightscout -- it could theoretically be extended to support writing to other sources like Sugarmate but it would take a lot of work, and it is debatable whether that's the right direction for tconnectsync to move towards. 

Maybe you could use tconnectsync, sending pump data to Nightscout, in addition to a separate system to bridge data from Nightscout to Sugarmate? This help article came up in a search, although it looks like it might have recently stopped working for newer versions of Sugarmate: https://help.sugarmate.io/en/articles/4673402-adding-a-nightscout-data-source

**Comment by ariana-wolfkiel** [comment]
Yes, we went through the SugarMate problem described in the article. Dexcom EOL'd the method SugarMate was using to pull the data directly from Dexcom. SugarMate previously had the Nightscout method but it had fallen out of favor since there was no latency getting BG directly from Dexcom. Then Dexcom broke the feed. So, the whole community either switched over to the Nightscout method or just did without (lots of pain for those unable to figure out Nightscout). Apparently that got Dexcom's attention since they very quickly negotiated a deal with SugarMate to allow a direct feed again. But the price of Dexcom's participation was that SugarMate had to agree to disable the Nightscout feed option in all future versions. Devil's bargain. Note that SugarMate is now owned by Tandem. They've owned it for over a year and done nothing with it (except play games with the Dexcom feed). 

That said, SugarMate is very popular, especially among the less-technical crowd who would have difficulty setting up Nightscout. The interface is simpler, cleaner, more intuitive, and I much prefer the scrolling. We like it because any caregiver (daycare, school med-tech, grandparent, etc.) can put in carbs, finger stick readings, boluses, notes, etc. and everyone else (parents) can see it. We could also (!!!) see the ControlIQ boluses that we have no visibility of since they only show up on the cell phone Tandem app - so less possibility of accidental stacking.

Until we can switch to Omnipod 5 (March?) our best option for Tandem X2 is to keep using SugarMate. tconnectsync would make capturing the carbs and boluses easier since the caregiver wouldn't have to manually put them in. Less problems with forgetting and incorrect time tags and accidental stacking.
